dev/com.ibm.ws.jca_fat_regr/fat/src/com/ibm/ws/jca/fat/regr/app/InboundSecurityTest.java

```diff
@@ -0,0 +1,938 @@
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.jca.fat.regr.app;
+
+import static org.junit.Assert.assertNotNull;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.ibm.websphere.simplicity.config.ServerConfiguration;
+import com.ibm.websphere.simplicity.log.Log;
+import com.ibm.ws.jca.fat.regr.util.JCAFATTest;
+
+import componenttest.annotation.ExpectedFFDC;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServerFactory;
+import suite.r80.base.jca16.RarTests;
+
+/**
+ *
+ */
+@RunWith(FATRunner.class)
+public class InboundSecurityTest extends JCAFATTest implements RarTests {
+    private final static Class<?> c = InboundSecurityTest.class;
+    private final String servletName = "InboundSecurityTestServlet";
+    private static ServerConfiguration originalServerConfig;
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        server = LibertyServerFactory.getLibertyServer("com.ibm.ws.jca.fat.regr", null, false);
+        originalServerConfig = server.getServerConfiguration().clone();
+        server.startServer("InboundSecurityTest.log");
+        server.waitForStringInLog("CWWKE0002I");
+        server.waitForStringInLog("CWWKZ0001I:.*fvtapp"); // Wait for application start.
+        server.waitForStringInLog("CWWKF0011I");
+        server.waitForStringInLog("CWWKS4104A"); // Wait for Ltpa keys to be generated
+    }
+
+    @AfterClass
+    public static void tearDown() throws Exception {
+        try {
+            server.stopServer("J2CA0677E", // EXPECTED
+                              "J2CA0668E", // EXPECTED
+                              "J2CA0669E: .*CallerPrincipalCallback", // EXPECTED
+                              "J2CA0670E: .*NonExistentRealm/JosephABCDEF", // EXPECTED
+                              "J2CA0671E", // EXPECTED
+                              "J2CA0673W: .*(Caller|Group|Password)(Principal|Validation)Callback", // EXPECTED
+                              "J2CA0674E: .*(null|Joseph) .*PasswordValidationCallback", // EXPECTED
+                              "J2CA0675E: The user name Joseph, provided by the PasswordValidationCallback, and the user name Susan, provided by the CallerPrincipalCallback, do not match", // EXPECTED
+                              "J2CA0676E: .*CallerPrincipalCallback", // EXPECTED
+                              "J2CA0678W: Group testGrpNonExistent" // EXPECTED
+            );
+        } finally {
+            server.updateServerConfiguration(originalServerConfig);
+        }
+    }
+
+    @Test
+    public void testCallerIdentityPropagationFromCallerPrincipalCallbackStartWork() throws Exception {
+        String testName = "testCallerIdentityPropagationFromCallerPrincipalCallbackStartWork";
```

Comment:
If `JCAFATTest` extends `FATServletClient` then a protected field called `testName` will automatically be created.  This is important to have because if we try to repeat these tests with Jakartaee10 then the method name will be changed to `testCallerIdentityPropagationFromCallerPrincipalCallbackStartWork_EE10` (or something like that).

Reply:
fixed